redwood/src/lib.rs

```diff
@@ -244,4 +244,34 @@ mod tests {
         // Verify message is what we put in originally
         assert_eq!("Rust is great 🦀", &plaintext);
     }
+
+    #[test]
+    fn test_encryption_missing_malformed_recipient_key() {
+        // Bad fingerprints can be: empty, empty string, or malformed
+        let bad_keys: Vec<(Vec<String>, &str)> = vec![
+            (
+                vec![],
+                "OpenPGP error: Invalid operation: \
+                Neither recipients, passwords, nor session key given",
+            ),
+            (
+                vec!["".to_string()],
+                "OpenPGP error: Malformed Cert: No data",
+            ),
+            (
+                vec!["DEADBEEF0123456".to_string()],
+                "OpenPGP error: unexpected EOF",
+            ),
+        ];
+        let tmp: NamedTempFile = NamedTempFile::new().unwrap();
+        for bad_key in bad_keys {
```

Comment:
One small nitpick that is hopefully useful for learning purposes: you can unpack the tuple while iterating, so this could be:
```rust
for (recipients, expected_err) in bad keys {
    ...
}
```

instead of having to refer to them as `.0` or `.1`.

Reply:
tis indeed an improvement, thank you for all the back and forth on this, it has been super helpful :)